### DISCUSSION TITLE
Cascade

### DISCUSSION THREAD

**Discussion by kamalkech** [typegoose/typegoose#607] [Q&A]
As i mention in the title, how can use cascade with this package ??

there is another way instead of  example bellow, same other configuration of model ??

```typescript
// First delete item
const res = await this.userModel.deleteOne({ _id }).exec();
// Cascade: then delete profile of user
await this.profileService.deleteByUser(_id);
return res;
```

**Comment by smolinari** [comment]
Firstly, there is a "Discussons" forum open for questions (edit: which this has now been moved to). Please use that in the future. Issues are reserved for bugs and suggestions of improvement. 

To answer your question, you could do what you've come up with or you could set up a remove hook. 
This article describes it: https://dev.to/kwabenberko/implementing-sql--like-cascades-in-mongoose-bap

Or, there are a couple of plugins built for this, like this one:
https://github.com/QuantumGlitch/mongoose-references-integrity-checker

I can't vouch for the plugin. I just found it looking for a possible plugin for such a feature. 

You can search the Mongoose plugins for "Cascade" to find the others. 
https://plugins.mongoosejs.io/

Scott

**Reply by kamalkech** [reply]
- https://dev.to/kwabenberko/implementing-sql--like-cascades-in-mongoose-bap: i already check this page before but how can integrate same logic using typegoose with nestjs, it s same implementation ?

- https://github.com/QuantumGlitch/mongoose-references-integrity-checker: it s not a good choose for real project and very low community, when check npm js 5 downloads weekly

- https://plugins.mongoosejs.io : there is no documentation about topic cascade using typegoose itself ?

thank you
